checker/safe_checking.ml

```diff
@@ -25,6 +25,18 @@ let pr_dirpath dp = str (DirPath.to_string dp)
 let genv = ref empty_env
 let get_env () = !genv
```

Comment:
We should carry this global ref functionally at some point.

Reply:
I'm taking care of this one, in fact almost the whole file can be refactored as well.